autoPyTorch/evaluation/train_evaluator.py

```diff
@@ -335,6 +397,68 @@ def eval_function(
         search_space_updates: Optional[HyperparameterSearchSpaceUpdates] = None,
         instance: str = None,
 ) -> None:
+    """
+    This closure allows the communication between the ExecuteTaFuncWithQueue and the
+    pipeline trainer (TrainEvaluator).
+
+    Fundamentally, smac calls the ExecuteTaFuncWithQueue.run() method, which internally
+    builds a TrainEvaluator. The TrainEvaluator builds a pipeline, stores the output files
+    to disc via the backend, and puts in the queue the performance result of the run.
```

Comment:
puts the performance result of the run in the queue

Reply:
Thanks, I have improved the docstring with this!